vllm/multimodal/utils.py

```diff
@@ -156,12 +146,9 @@ def _load_video_from_bytes(b: bytes, num_frames: int = 32):
 
 
 def _load_video_from_data_url(video_url: str):
-    # Only split once and assume the second part is the base64 encoded image
-    frames_base64 = video_url.split(",")[1:]
-    return np.stack([
-        load_video_frames_from_base64(frame_base64)
-        for frame_base64 in frames_base64
-    ])
```

Comment:
I think we should update `encode_video_base64` too, because it's encoding video frames to multiple pil.Image.

Reply:
Hmm, currently this function is only used in tests where the URL is assumed to be in `video/jpeg` format. I think there's not much reason to change it unless we intend users to import this function.

Reply:
Hmm, but the `test_video.py` entrypoint test is failing becasue decord is not compatiable with PIL.Image:
```
tests/entrypoints/openai/test_video.py::test_single_chat_session_video_base64encoded[http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ElephantsDream.mp4-llava-hf/llava-onevision-qwen2-0.5b-ov-hf] INFO:     127.0.0.1:41958 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/opt/conda/envs/vllm/lib/python3.10/site-packages/starlette/_utils.py", line 76, in collapse_excgroups
    yield
  File "/opt/conda/envs/vllm/lib/python3.10/site-packages/starlette/middleware/base.py", line 186, in __call__
    async with anyio.create_task_group() as task_group:
  File "/opt/conda/envs/vllm/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 815, in __aexit__
    raise BaseExceptionGroup(
exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  ...
  File "/kaggle/working/vllm/vllm/entrypoints/utils.py", line 54, in wrapper
    return handler_task.result()
  File "/kaggle/working/vllm/vllm/entrypoints/openai/api_server.py", line 371, in create_chat_completion
    generator = await handler.create_chat_completion(request, raw_request)
  File "/kaggle/working/vllm/vllm/entrypoints/openai/serving_chat.py", line 165, in create_chat_completion
    ) = await self._preprocess_chat(
  File "/kaggle/working/vllm/vllm/entrypoints/openai/serving_engine.py", line 485, in _preprocess_chat
    mm_data = await mm_data_future
  File "/kaggle/working/vllm/vllm/entrypoints/chat_utils.py", line 492, in all_mm_data
    items = await asyncio.gather(*self._items)
  File "/kaggle/working/vllm/vllm/multimodal/utils.py", line 281, in async_get_and_parse_video
    video = await async_fetch_video(video_url)
  File "/kaggle/working/vllm/vllm/multimodal/utils.py", line 187, in async_fetch_video
    video = _load_video_from_data_url(video_url)
  File "/kaggle/working/vllm/vllm/multimodal/utils.py", line 151, in _load_video_from_data_url
    return load_video_from_base64(video_base64)
  File "/kaggle/working/vllm/vllm/multimodal/utils.py", line 385, in load_video_from_base64
    return _load_video_from_bytes(base64.b64decode(video))
  File "/kaggle/working/vllm/vllm/multimodal/utils.py", line 132, in _load_video_from_bytes
    vr = decord.VideoReader(video_path, num_threads=1)
  File "/opt/conda/envs/vllm/lib/python3.10/site-packages/decord/video_reader.py", line 51, in __init__
    self._handle = _CAPI_VideoReaderGetVideoReader(
  File "/opt/conda/envs/vllm/lib/python3.10/site-packages/decord/_ffi/_ctypes/function.py", line 173, in __call__
    check_call(_LIB.DECORDFuncCall(
  File "/opt/conda/envs/vllm/lib/python3.10/site-packages/decord/_ffi/base.py", line 78, in check_call
    raise DECORDError(err_str)
decord._ffi.base.DECORDError: [14:23:24] /github/workspace/src/video/video_reader.cc:270: [BytesIO] Failed to measure duration/frame-count due to broken metadata.
FAILED
```

Reply:
Oh, that's a problem... let me see what I can do.

Reply:
I have added a fallback that directly uses `load_image_from_base64` instead of defining a separate function that's essentially identical to it.